modules/search-pipeline-common/src/main/java/org/opensearch/search/pipeline/common/SortResponseProcessor.java

```diff
@@ -0,0 +1,212 @@
+     * @return sortOrder
+     */
+    public SortOrder getSortOrder() {
+        return sortOrder;
+    }
+
+    @Override
+    public String getType() {
+        return TYPE;
+    }
+
+    @Override
+    public SearchResponse processResponse(SearchRequest request, SearchResponse response) throws Exception {
+        SearchHit[] hits = response.getHits().getHits();
+        for (SearchHit hit : hits) {
+            Map<String, DocumentField> fields = hit.getFields();
+            if (fields.containsKey(sortField)) {
+                DocumentField docField = hit.getFields().get(sortField);
+                if (docField == null) {
+                    throw new IllegalArgumentException("field [" + sortField + "] is null, cannot sort.");
+                }
+                hit.setDocumentField(targetField, new DocumentField(targetField, getSortedValues(docField.getValues())));
+            }
+            if (hit.hasSource()) {
+                BytesReference sourceRef = hit.getSourceRef();
+                Tuple<? extends MediaType, Map<String, Object>> typeAndSourceMap = XContentHelper.convertToMap(
+                    sourceRef,
+                    false,
+                    (MediaType) null
+                );
+
+                Map<String, Object> sourceAsMap = typeAndSourceMap.v2();
+                if (sourceAsMap.containsKey(sortField)) {
+                    Object val = sourceAsMap.get(sortField);
+                    if (val instanceof List) {
+                        @SuppressWarnings("unchecked")
+                        List<Object> listVal = (List<Object>) val;
+                        sourceAsMap.put(targetField, getSortedValues(listVal));
+                    }
+                    XContentBuilder builder = XContentBuilder.builder(typeAndSourceMap.v1().xContent());
+                    builder.map(sourceAsMap);
+                    hit.sourceRef(BytesReference.bytes(builder));
+                }
+            }
+        }
+        return response;
+    }
+
+    private <T extends Comparable<T>> List<Object> getSortedValues(List<Object> values) {
+        List<T> comparableValues = new ArrayList<>(values.size());
+        for (Object obj : values) {
+            if (obj == null) {
+                throw new IllegalArgumentException("field [" + sortField + "] contains a null value.]");
+            } else if (Comparable.class.isAssignableFrom(obj.getClass())) {
+                @SuppressWarnings("unchecked")
+                T comp = (T) obj;
+                comparableValues.add(comp);
+            } else {
+                throw new IllegalArgumentException(
+                    "field [" + sortField + "] of type [" + obj.getClass().getName() + "] is not comparable.]"
+                );
+            }
+        }
+        return comparableValues.stream()
+            .sorted(sortOrder.equals(SortOrder.ASCENDING) ? Comparator.naturalOrder() : Comparator.reverseOrder())
+            .collect(Collectors.toList());
+    }
```

Comment:
We can split the methods into two here. Reasons:
1. More readability
2. Can use Collections.sort which in turns sort the list in-place and same like Ingest pipeline's split processor
```suggestion
    private List<? extends Comparable<Object>> getComparableValues(List<Object> values) {
    List<Comparable<Object>> comparableValues = new ArrayList<>(values.size());
    for (Object obj : values) {
        if (obj == null) {
            throw new IllegalArgumentException("field [" + sortField + "] contains a null value.");
        } else if (obj instanceof Comparable) {
            @SuppressWarnings("unchecked")
            Comparable<Object> comp = (Comparable<Object>) obj;
            comparableValues.add(comp);
        } else {
            throw new IllegalArgumentException(
                "field [" + sortField + "] of type [" + obj.getClass().getName() + "] is not comparable."
            );
        }
    }
    return comparableValues;
}

private List<? extends Comparable<Object>> sortValues(List<? extends Comparable<Object>> values) {
    List<Comparable<Object>> copy = new ArrayList<>(values);
    if (sortOrder.equals(SortOrder.ASCENDING)) {
        Collections.sort(copy);
    } else {
        Collections.sort(copy, Collections.reverseOrder());
    }
    return copy;
}
```

Reply:
I actually did have it as two separate methods, but since the second method can be written as a one-liner with streams I combined them.

I agree that including a ternary operator in the middle of a stream strains readability, though, so I'll go back to the two-phase method.

Reply:
Actually, it's not that simple.  While you can sort in-place, the return type necessary is `List<Object>` to be used in the call to the `DocumentField` constructor.  And you can't change the type of an object in-place.

I'll still separate the methods for readability, but there will still be a stream at the end to do the typecast.

Reply:
Can you access them something like the below?
```
List<? extends Comparable<Object>> values = getComparableValues(docField.getValues());
List<? extends Comparable<Object>> sortedValues = sortValues(values);
```

Reply:
> Can you access them something like the below?

`List<? extends Comparable<Object>> sortedValues = ...` does not compile with `new DocumentField(targetField, sortedValues)`.